artiq/firmware/runtime/rtio_mgt.rs

```diff
@@ -349,23 +353,45 @@ fn async_error_thread(io: Io) {
             io.until(|| csr::rtio_core::async_error_read() != 0).unwrap();
             let errors = csr::rtio_core::async_error_read();
             if errors & ASYNC_ERROR_COLLISION != 0 {
-                error!("RTIO collision involving channel {}",
-                       csr::rtio_core::collision_channel_read());
+                let channel = csr::rtio_core::collision_channel_read();
+                error!("RTIO collision error involving channel {} ({})",
+                       channel, config_channel_name(channel));
+                
             }
             if errors & ASYNC_ERROR_BUSY != 0 {
-                error!("RTIO busy error involving channel {}",
-                       csr::rtio_core::busy_channel_read());
+                let channel = csr::rtio_core::busy_channel_read();
+                error!("RTIO busy error involving channel {} ({})",
+                       channel, config_channel_name(channel));
             }
             if errors & ASYNC_ERROR_SEQUENCE_ERROR != 0 {
-                error!("RTIO sequence error involving channel {}",
-                       csr::rtio_core::sequence_error_channel_read());
+                let channel = csr::rtio_core::sequence_error_channel_read();
+                error!("RTIO sequence error involving channel {} ({})",
+                       channel, config_channel_name(channel));
             }
             SEEN_ASYNC_ERRORS = errors;
             csr::rtio_core::async_error_write(errors);
         }
     }
 }
 
+fn config_channel_name(ch_number: u16) -> String {
+    let mut ch_name = String::from("unknown");
+    config::read_str("channel_names", |r| match r {
+        Ok(ch_names) => {
+            let split_1 = ch_names.split(",");
+            for s in split_1 {
+                let mut split_2 = s.split(":");
+                if split_2.next().unwrap() == format!("{}", ch_number) {
```

Comment:
Use ``to_string()`` (no need for the heavy format machinery) and store the result at the beginning of the function (no need to allocate/deallocate memory at each iteration).

Reply:
![Screenshot from 2022-08-02 17-37-46](https://user-images.githubusercontent.com/48083317/182347343-1a561e9a-6f89-4068-b382-129fc25757a6.png)
 I've tried to put the result at the beginning. It seems impossible to avoid escaping.

Reply:
I was talking about ``format!("{}", ch_number)``